File: 206-reverse-linked-list/206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *curr = head;
    struct ListNode *prev = NULL;
    struct ListNode *following = head;
    
    while (curr != NULL) {
        following = following->next;
        curr->next = prev;
        prev = curr;
        curr = following;
    }
    return prev;
}

File: 206-reverse-linked-list/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head 
        following = head
        prev = None 
    
        while curr:
            following = following.next 
            curr.next = prev 
            prev = curr
            curr = following 

        return prev 
        